src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnInit,
+  OnDestroy,
+  ElementRef,
+  TemplateRef
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
+import 'rxjs/add/observable/zip';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import {
+  SkyInlineFormButton,
+  SkyInlineFormCloseArgs,
+  SkyInlineFormType,
+  SkyInlineFormConfig,
+  SkyInlineFormButtonConfig
+} from './types';
+
+import {
+  SkyInlineFormAdapterService
+} from './inline-form-adapter.service';
+
+import {
+  skySlideDissolve
+} from './animations';
+
+@Component({
+  selector: 'sky-inline-form',
+  templateUrl: './inline-form.component.html',
+  styleUrls: ['./inline-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [ skySlideDissolve ]
+})
+export class SkyInlineFormComponent implements OnInit, OnDestroy {
+
+  @Output()
+  public close = new EventEmitter<SkyInlineFormCloseArgs>();
+
+  @Input()
+  public config: SkyInlineFormConfig;
+
+  @Input()
+  public template: TemplateRef<any>;
+
+  @Input()
+  set showForm(value: boolean) {
+    this._showForm = value;
+
+    if (value) {
+      window.setTimeout(() => this.adapter.applyAutofocus(this.elementRef));
+    }
+  }
+
+  get showForm() {
+    return this._showForm;
+  }
+
+  public buttons: SkyInlineFormButton[];
+
+  private _showForm: boolean = false;
+
+  constructor(
+    private adapter: SkyInlineFormAdapterService,
+    private elementRef: ElementRef,
+    private resourcesService: SkyLibResourcesService
+  ) {}
+
+  public ngOnInit(): void {
+    if (this.isValidCustomConfig(this.config)) {
+      this.buttons = this.getCustomButtons(this.config.buttons);
+    } else {
+      this.getPresetButtons().subscribe((buttons: SkyInlineFormButton[]) => {
```

Comment:
Does this need a `takeUntil`? It looks like you're using observables in a "promisey" way in the `getPresetButtons` method; maybe it could return a simple `Promise<SkyInlineFormButton[]>` and you wouldn't have to worry about subscriptions?